tests/api_workflow/test_api_workflow_collaboration.py

```diff
@@ -0,0 +1,17 @@
+from tests.api_workflow.mocked_api_workflow_client import MockedApiWorkflowSetup, MockedApiWorkflowClient
+
+
+class TestApiWorkflowDatasets(MockedApiWorkflowSetup):
+
+    def setUp(self) -> None:
+        self.api_workflow_client = MockedApiWorkflowClient(token="token_xyz")
+
+    def test_share_empty_dataset(self):
+        self.api_workflow_client.share_dataset(dataset_id="some-dataset-id", user_emails=[])
+
+    def test_share_dataset(self):
+        self.api_workflow_client.share_dataset(dataset_id="some-dataset-id", user_emails=["someone@something.com"])
```

Comment:
where is the validation?

Reply:
Could you please explain what you mean with validation? I just followed the example tests we have for the dataset creation (see https://github.com/lightly-ai/lightly/blob/85e21e6008a7683ead1f6f7aaf6977412b9d7ddd/tests/api_workflow/test_api_workflow_datasets.py#L8)

Reply:
I might not be the best to explain this :D but as this endpoint is mocked we are essentially testing what we mocked and nothing else?
What I normally would do on the API is "share it", "get shared back", "check if shared is what i wanted to share it with", "share it with something else", "check if its changed".
Maybe you can mock the endpoint from the test itself and check that the API function was called with certain params but also that seems a bit ridiculous to do. @MalteEbner help :D

Reply:
@japrescott I assume you mean something like this:
https://gist.github.com/hectorcanto/ecc42cf5d5ad4ea309f51d21337d537c

Asking @michal-lightly how to do this best.

Reply:
I agree with Jeremy. Basically the test should do what is explained in the docstring (`lightly/api/api_workflow_collaboration.py:30`).

Regarding mocking, I don't know how APIWorkflowClient is mocked, so I can say just general things: Test the functionality that you added. If it has some expensive call that is mocked, you can check that the call has been made with the right parameters.

Reply:
We discussed this and Malte will organize a session on how to do the mocking properly.